packages/worker/src/isolate-builder.ts

```diff
@@ -0,0 +1,151 @@
+import path from 'path';
+import os from 'os';
+import fs from 'fs-extra';
+import dedent from 'dedent';
+import ivm from 'isolated-vm';
+import webpack from 'webpack';
+import { ActivityOptions, validateActivityOptions } from '@temporalio/workflow';
+import { Logger } from './logger';
+
+export class WorkflowIsolateBuilder {
+  constructor(
+    public readonly logger: Logger,
+    public readonly nodeModulesPath: string,
+    public readonly workflowsPath: string,
+    public readonly activities: Map<string, Record<string, any>>,
+    public readonly activityDefaults: ActivityOptions
+  ) {}
+
+  public async build(): Promise<ivm.Isolate> {
+    const distDir = await fs.mkdtemp(path.join(os.tmpdir(), 'temporal-bundler-out-'));
+    this.logger.info('Building Workflow code for Worker isolate', { distDir });
+    try {
+      const prebuildDir = path.join(distDir, 'prebuild');
+      await fs.mkdir(prebuildDir, { recursive: true });
+      await this.registerActivities(prebuildDir);
+      const entrypointPath = path.join(prebuildDir, 'main.js');
+      const workflows = await this.findWorkflows();
+      await this.genEntrypoint(entrypointPath, workflows);
+      await this.bundle(entrypointPath, distDir);
+      const code = await fs.readFile(path.join(distDir, 'main.js'), 'utf8');
+      const snapshot = ivm.Isolate.createSnapshot([{ code }]);
+      return new ivm.Isolate({ snapshot });
+    } finally {
+      await fs.remove(distDir);
+    }
+  }
+
+  protected async findWorkflows(): Promise<string[]> {
+    const files = await fs.readdir(this.workflowsPath);
+    return files.filter((f) => path.extname(f) === '.js').map((f) => path.basename(f, path.extname(f)));
+  }
+
+  protected async genEntrypoint(target: string, workflows: string[]): Promise<void> {
+    let code = dedent`
+      const init = require('@temporalio/workflow/lib/init');
+      const internals = require('@temporalio/workflow/lib/internals');
+      internals.state.activityDefaults = ${JSON.stringify(this.activityDefaults)};
+
+      const workflows = {};
+      export { init, internals, workflows };
+    `;
+    for (const wf of workflows) {
+      code += `workflows[${JSON.stringify(wf)}] = require(${JSON.stringify(path.join(this.workflowsPath, wf))});\n`;
+    }
+    await fs.writeFile(target, code);
```

Comment:
Might be nice to avoid writing files to disk, but not sure if you can.

https://webpack.js.org/configuration/entry-context/#entry

Is unclear if you can pass file contents or some already-in-memory thing or if it always has to be on disk